### ISSUE TITLE
Provide default_page and layout properties in content GET

### ISSUE THREAD

**Issue by buchi** [plone/plone.restapi#411]
Needed to be able to render the default page or the correct view in the frontend.

Example:
```
{
    "@id": "http://localhost:8080/Plone",
    "@type": "Plone Site",
    "default_page": "http://localhost:8080/Plone/front-page",
    ...
}
```

or make it an expandable component?

```
{
    "@id": "http://localhost:8080/Plone",
    "@type": "Plone Site",
    "@components": {
        "properties": {
            "@id": "http://localhost:8080/Plone/@properties",
            "items": [
                 "default_page": "http://localhost:8080/Plone/front-page"
            ]
        },
    ...
}
```

See also #126

@tisto @sneridagh @ebrehault opinions?

**Comment by robgietema** [comment]
I really dislike the whole default_page method it confuses end users. I suggest we will go for collective.folderishtypes which makes a lot more sense. see also #427

**Comment by davisagli** [comment]
Personally I think that while default pages are a bit confusing, they have their uses. In a world with default pages, an editor can prepare a major edit of a section landing page as a separate page, and then easily activate it. In a world without default pages they would need to move the entire subtree into the new landing page, delete the old one, and rename the new one. The 1st and 3rd operations would take a lot of time with a big subtree, and there would be some time in between where the landing page is missing.

Anyway, this probably isn't the best place to have that argument. In our current world where default pages exist, it would be useful to have that info in the API.

**Comment by datakurre** [comment]
How about portal root?

**Comment by zopyx** [comment]
Like default pages or dislike them..they are a core feature of Plone and in some case you need to use them and therefore they should have some support in the REST API - meanwhile I wrote my own endpoint inside my provisioning API.

**Comment by wobsta** [comment]
@zopyx Could you share some code you created to access the default_page property of a folder? I'm in need to set this property once per folder (nothing else at the moment) in some import code I'm working on ...

**Comment by zopyx** [comment]
https://docs.plone.org/develop/plone/content/dynamic_views.html#changing-default-view-programmatically

**Comment by wobsta** [comment]
While trying to implement something to get my work done, I unintentionally found a way to "set" the default page when uploading a new document with the REST API without needing to patch anything at all. It is rather easy: You can upload (by POST) a new document and use the name `index_html` (which must not be in use already). Due to this name it becomes the default view of the folder automatically. You can than immediately rename the document (by PATCH) and it still remains flagged as the default page. Works fine for me and is all I need. Maybe others might want to use that solution as well.

**Comment by zopyx** [comment]
The ticket is about changing the layout and setting the default page.

**Comment by wobsta** [comment]
> Am 27.07.2018 um 16:45 schrieb Andreas Jung <notifications@github.com>:
> 
> The ticket is about changing the layout and setting the default page.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 


@zopyx Don't mind, but your ticket #499 was marked as a duplicate and is all about setting the default page. My comment would probably have been a better fit to this issue, although it has been closed and marked as duplicate by @tisto several weeks ago. Anyway, I just posted my tentative solution ... sorry for the noise.

-- 
by  _ _      _    Dr. André Wobst, Amselweg 22, 85716 Unterschleißheim
   / \ \    / )   contact@wobsta.de, http://www.wobsta.de/
  / _ \ \/\/ /    Tel: (089) 5480 3394, Fax: (089) 5480 3396
 (_/ \_)_/\_/     Handy: (0160) 9380 4217

**Comment by tisto** [comment]
@wobsta right. This is Plone standard behavior. Good to know this solves your problem. Thanks for sharing it! I'd still be ok with implementing the default_page and layout in plone.restapi. Especially in plone-react it would be handy to being able to set the layout of a particular object. We use that a lot at kitconcept...So if anybody is willing to implement this, I'd be more than happy to help with pointers and reviewing...

**Comment by tisto** [comment]
@buchi making default_page and layout an expandable might be overkill IMHO. Maybe just get/post them as regular attributes? As always, we would have to live with possible name collisions. Though, I guess I'd be ok with that. Opinions?

**Comment by robgietema** [comment]
@davisagli that seems to be a versioning workflow (can be done with plone.app.iterate?), using the default page for that is just a convenient hack.

@datakurre the portal_root issue is fixed with this PLIP https://github.com/plone/Products.CMFPlone/issues/2454

**Comment by sauzher** [comment]
@tisto 
> @sauzher didn't you signed the Plone contributor agreement or what's the reason you can not push your branch here.

I think I've done (ages ago actually). Will do again. Tnx!

**Comment by thomasmassmann** [comment]
@sauzher , can you please create a PR so we can integrate this. Also, we should add the same functionality for the deserializer.

**Comment by sauzher** [comment]
@thomasmassmann 
is this ok? 
https://github.com/plone/plone.restapi/pull/983
